src/util/ivmFactory.js

```diff
@@ -95,7 +95,7 @@ async function createIvm(
       const isObject = (o) => Object.prototype.toString.call(o) === '[object Object]';
 
       var metadata = function(event) {
-        const eventMetadata = event ? eventsMetadata[event.messageId] || {} : {};
+        const eventMetadata = event ? JSON.parse(JSON.stringify(eventsMetadata[event.messageId] || {})) : {};
```

Comment:
any reason for not using lodash deepCopy?

Reply:
I would suggest we don't pass the metadata as it is even after the copy. We cherry pick the data which we wanna send in through keys and just return that object. 

Wdyt ?

Reply:
incase we want to clone -> structuredClone is faster and native no need to use lodash cloneDeep

Reply:
didn't cherry pick the data because we have to parse transformation codes and validate the required fields

Reply:
did an online js benchmark test using actual metadata object:

- structuredClone: 123K ops/s ± 2.08%
- JSON way:           186K ops/s ± 0.66%